qiskit/providers/backend.py

```diff
@@ -229,3 +240,435 @@ class can handle either situation.
+    are added. For example ``backend.configuration().n_qubits`` is accessible
+    from ``backend.num_qubits`` now.
+
+    The ``options`` attribute of the backend is used to contain the dynamic
+    user configurable options of the backend. It should be used more for
+    runtime options that configure how the backend is used. For example,
+    something like a ``shots`` field for a backend that runs experiments which
+    would contain an int for how many shots to execute.
+
+    If migrating a provider from :class:`~qiskit.providers.BackendV1` or
+    :class:`~qiskit.providers.BaseBackend` one thing to keep in mind is for
+    backwards compatibility you might need to add a configuration method that
+    will build a :class:`~qiskit.providers.models.BackendConfiguration` object
+    and :class:`~qiskit.providers.models.BackendProperties` from the attributes
+    defined in this class for backwards compatibility.
+    """
+
+    version = 2
+
+    def __init__(
+        self,
+        provider,
+        name: str = None,
+        description: str = None,
+        online_date: datetime.datetime = None,
+        backend_version: str = None,
+        **fields,
+    ):
+        """Initialize a BackendV2 based backend
+
+        Args:
+            provider (Provider): A backwards reference to the
+                :class:`~qiskit.transpiler.Provider` object that the backend
+                is from
+            name: An optional name for the backend
+            description: An optional description of the backend
+            online_date: An optional datetime the backend was brought online
+            backend_version: An optional backend version string. This differs
+                from the :attr:`~qiskit.providers.BackendV2.version` attribute
+                as :attr:`~qiskit.providers.BackendV2.version` is for the
+                abstract :attr:`~qiskit.providers.Backend` abstract interface
+                version of the object while ``backend_version`` is for
+                versioning the backend itself.
+
+        Raises:
+            AttributeError: If a field is specified that's outside the backend's
+                options
+        """
+
+        self._options = self._default_options()
+        self._provider = provider
+        self._coupling_map = None
+        if fields:
+            for field in fields:
+                if field not in self._options.data:
+                    raise AttributeError("Options field %s is not valid for this backend" % field)
+            self._options.update_config(**fields)
+        self._basis_gates_all = None
+        self.name = name
+        self.description = description
+        self.online_date = online_date
+        self.backend_version = backend_version
+
+    @property
+    def instructions(self) -> List[Tuple[Instruction, Tuple[int]]]:
+        """A list of Instruction tuples on the backend of the form ``(instruction, (qubits)``"""
```

Comment:
It seems like naming for operations and instructions are opposite. 
- operations:= Instruction
- instructions:= (Instruction, qubits)

This is because we have already defined `Instruction` class that is agnostic to qubit and we cannot easily change its name?

Reply:
Yeah, this is the direction @ajavadia wants to move everything but it's not an easy transition to make because as you say `Instruction` is widely used as the class which is independent of the arguments. I went with the new convention here just so we don't have to update it after everything else is migrated eventually.

Reply:
yeah that is also how "instruction" is defined in `InstructionScheduleMap` (i.e. the gate plus the qubit it acts on). 
After `Operation` lands in Qiskit (https://github.com/Qiskit/qiskit-terra/pull/7087) and different Operations are defined (gates, measure, reset, clifford, unitary, etc.), then I think the type here can become `List[Tuple[Operation, Tuple[int]]]`